fix: allow Redis only and fix Redis password use in BROKER_URL

Kampfmoehre opened the pull request:
This allows to skip the deployment of RabbitMQ and therefore skip the deployment of the Sentry worker that connects to it.

This also fixes the Sentry Redis Worker, so that the correct BROKER_URL is used, when Redis is configured to connect with password.

This also updates the Sentry configmap to set the Broker URL to a Redis connection string, if RabbitMQ is disabled.

Kampfmoehre committed:
fix: either deploy redis or rabbitmq worker

Adds an enabled setting for RabbitMQ and Redis so the user can choose
what to use.

Kampfmoehre committed:
fix: set correct BROKER_URL when using Redis

Updates Sentry configmap to set redis as Broker URL if no RabbitMQ is
deployed.

Kampfmoehre committed:
fix: set correct Broker URL for Redis worker

Updates Redis worker deployment to include the password when set

Kampfmoehre committed:
chore: updates chart version

Kampfmoehre committed:
fix: remove wrong eq from if

Kampfmoehre committed:
style: add empty line at end of worker deployment

Kampfmoehre committed:
fix: use Redis Broker URL in cron, if no RabbitMQ

When RabbitMQ is disabled, use Redis connection string as BROKER_URL
for the Sentry cron.

Mokto commented:
Amazing! Thanks.

I'll put my comments right away

Mokto commented on sentry/templates/configmap-sentry.yaml:
```diff
@@ -161,7 +161,16 @@ data:
+    {{- if .Values.rabbitmq.enabled }}
     BROKER_URL = os.environ.get("BROKER_URL", "amqp://{{ .Values.rabbitmq.username }}:{{ .Values.rabbitmq.password }}@{{ .Values.rabbitmq.host }}:5672//")
+    {{- else if .Values.redis.enabled }}
+      {{- if .Values.redis.password }}
```
This is repeated, could you use a template helper instead?

Thanks.

Kampfmoehre replied:
I changed this, since Redis is always enabled. Since I'm fairly new to helm charts, I'm not sure how to write helpers yet. But I could take a look at it.

Mokto commented on sentry/templates/deployment-sentry-cron.yaml:
```diff
@@ -61,7 +61,15 @@ spec:
           - name: BROKER_URL
+            {{- if .Values.rabbitmq.enabled }}
             value: amqp://{{ .Values.rabbitmq.username }}:{{ .Values.rabbitmq.password }}@{{ .Values.rabbitmq.host }}:5672
+            {{- else if .Values.redis.enabled }}
```
`{{- else }}` is enough here. One of those is required anyways

Kampfmoehre replied:
fixed

Mokto commented on sentry/templates/deployment-sentry-worker-redis.yaml:
```diff
@@ -1,3 +1,4 @@
+{{- if .Values.redis.enabled }}
```
`if not .Values.rabbitmq.enabled`

Kampfmoehre replied:
fixed

Mokto commented on sentry/values.yaml:
```diff
@@ -136,11 +136,15 @@ postgres:
 
 redis:
+  # either use RabbitMQ or Redis as broker
+  enabled: false
```
This implies that Redis might not be needed.
Redis is always needed.

You can remove that and put `if not .Values.rabbitmq.enabled` as a condition.

Kampfmoehre replied:
Removed

Kampfmoehre committed:
refactor: simplify logic, since Redis is required

Simplifies some if/else conditions, since Redis is always enabled.
Also remove redis.enabled setting from values.

Mokto commented:
Thanks ! That's perfect. Could you just activate rabbitmq per default, please ?

Kampfmoehre committed:
fix: active RabbitMQ per default

Kampfmoehre commented:
Of course.
Btw: I thought about putting password in secrets, did you start any work on this yet?

Mokto commented:
No, not yet ! Go for it ;) 

I'll merge your PR in the early evening.

Thanks for your work !
